conans/client/cmd/export.py

```diff
@@ -15,56 +15,52 @@
 from conans.model.scm import detect_repo_type
 from conans.paths import CONANFILE
 from conans.search.search import search_recipes, search_packages
-from conans.util.files import is_dirty, load, mkdir, rmdir, save, set_dirty, remove
+from conans.util.files import is_dirty, load, rmdir, save, set_dirty, remove
 from conans.util.log import logger
 
 
-def export_alias(reference, target_reference, cache, output):
-    if reference.name != target_reference.name:
-        raise ConanException("An alias can only be defined to a package with the same name")
+def export_alias(package_layout, target_reference, output, revisions_enabled):
     conanfile = """
 from conans import ConanFile
 
 class AliasConanfile(ConanFile):
     alias = "%s"
-""" % target_reference.full_repr()
+""" % target_reference
```

Comment:
This is changing behavior, the previous behavior assumed you can make an alias to an exact reference. Check with @lasote this change.

Reply:
The broken behavior is here (https://github.com/conan-io/conan/pull/4496/files/493d118368d06b83ac09506793e7bf3dad21ba8f#diff-633cc9158b73bec9defbb8eb390ba040R961), where I'm doing `str(ref)` instead of `ref.full_repr`. 

Good catch! I didn't intend to change the behavior, now I'm going to revert this and add an alias test using revisions.

Reply:
Great!

Reply:
There are alias tests using revision, you just need to run them.

Reply:
@jgsogo I think this is still broken, this PR should run with revisions enabled.

Reply:
Running with revisions, let's see what happens.

Reply:
Run again, no changes (just merging from `develop`) and it works.

Reply:
Now I see what is happening. There is a small change below, the argument to this function is now a ``str`` instead of a ``ConanFileReference``, because it was converted to ``ref.full_repr()`` when the function was called.

I'd say it is better to pass around domain objects as ``ConanFileReference``, and convert them in the edges, in this case, just before being written to a text file.

Reply:
Yes, it is better the other way around, here I preserved the function signature (`target_reference`) and pass the proper instead of doing it the other way. I change it, I agree it is better.